### DISCUSSION TITLE
"Action '-DFOREGROUND' failed" for apache container?

### DISCUSSION THREAD

**Discussion by jonnyparris** [nextcloud/all-in-one#1219] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Try to run aio container

### Expected behavior 
Nextcloud Apace container runs as expected on port 11000.

### Actual behavior 
All requests are rejected, and I see these logs on start up:
```
[Sat Oct 01 22:02:28.713549 2022] [core:warn] [pid 19:tid 139767844851008] AH00111: Config variable ${APACHE_MAX_SIZE} is not defined
[Sat Oct 01 22:02:28.713705 2022] [core:warn] [pid 19:tid 139767844851008] AH00111: Config variable ${APACHE_MAX_TIME} is not defined
[Sat Oct 01 22:02:28.713711 2022] [core:warn] [pid 19:tid 139767844851008] AH00111: Config variable ${APACHE_MAX_TIME} is not defined
AH00526: Syntax error on line 33 of /etc/apache2/sites-enabled/nextcloud.conf:
LimitRequestBody requires a non-negative integer.
Action '-DFOREGROUND' failed.
The Apache error log may have more information.
```

### Host OS 
Ubuntu 20.04.2 LTS

#### Nextcloud AIO version 
2.0.3

#### Current channel 
latest

#### Other valuable info 
It was working fine up until 5 days ago 🤷🏾

**Comment by szaimen** [comment]
Hi, can you please post the output of `sudo docker inspect nextcloud-aio-apache`? Thanks!

**Comment by jonnyparris** [comment]
```
[redacted]
```

**Comment by szaimen** [comment]
Thanks! Can you also please run `docker inspect nextcloud-aio-mastercontainer` and post the ouput here?

To me it looks like somehow the apache container was updated but the mastercontainer not. Not sure how this could happen though since we require to update the mastercontainer before all other containers.... Is maybe a mastercontainer update available from the AIO interface?

**Comment by jonnyparris** [comment]
```
[redacted]

```

**Comment by jonnyparris** [comment]
No update available as far as I can see from the aio ui

**Comment by szaimen** [comment]
Seems like the image is indeed up-to-date. Can you try to restart the container from the AIO interface and afterwards post the output of `sudo docker inspect nextcloud-aio-apache` again? Also can you post the output of `sudo docker exec -it nextcloud-aio-mastercontainer cat php/containers.json`?

**Comment by jonnyparris** [comment]
Not sure how to restart the container from the aio interface because I hit this screen: 
<img width="632" alt="image" src="https://user-images.githubusercontent.com/6400000/193447759-a192b4ef-2862-49c5-9579-2fe6ce08651a.png">
and then once I've stopped the container via the commandline, it just says that the container is "stopped" in the interface, where clicking it just takes me to its logs.

Here's the output of `php/containers.json`:
```
[redacted]
```

**Comment by szaimen** [comment]
> Not sure how to restart the container from the aio interface because I hit this screen:

See https://github.com/nextcloud/all-in-one#how-to-easily-log-in-to-the-aio-interface

Restarting the containers from the aio interface is still necessary. Please try that again and see if it works afterwards.

**Comment by jonnyparris** [comment]
> See https://github.com/nextcloud/all-in-one#how-to-easily-log-in-to-the-aio-interface

...but I can't access that route because of this bug ☹️ 
Is restarting via the commandline not the same? / Is there a commandline equivalent besides `docker restart`?

**Comment by jonnyparris** [comment]
Actually I just stopped the apache container directly and stoppped and restarted all the containers via the aio ui and now it's working 🥳  - thanks for the fast support over the last few days!

Now the main questions are why did it happen? and how can it be prevented from happening again? The seamless auto-update is kind of the whole USP of the nc-aio setup but this threatens to be a recurring showstopper with each auto-version bump.
On a related note, is there a recommended way to setup alerts for when services go down? (I only found out about this one almost a week after the apache server became unresponsive)

**Comment by szaimen** [comment, marked as answer]
I had a look at the code in question and the only way that I seee how this could have happpened is if you update the containers externally e.g. using watchtower...

**Reply by jonnyparris** [reply]
aha, now that is feasible! I'll confirm when I get a chance

**Reply by jonnyparris** [reply]
confirming Watchtower was enabled and the more-than-likely culprit.

Thanks for the support!
